src/Tempest/Http/src/RouteDiscovery.php

```diff
@@ -24,21 +28,31 @@ public function discover(ClassReflector $class): void
                 continue;
             }
 
-            $this->routeConfig->addRoute($method, $routeAttribute);
+            $routeAttribute->setHandler($method);
+
+            $this->routeConfigConstructor->addRoute($routeAttribute);
+        }
+    }
+
+    #[EventHandler(KernelEvent::BOOTED)]
+    public function finishDiscovery(): void
+    {
+        if ($this->routeConfigConstructor->hasRoutes()) {
+            $this->routeConfig->apply($this->routeConfigConstructor->toRouteConfig());
         }
     }
```

Comment:
Previously the method `prepareMatchingRegexes` would ensure the tree is build and applied into the routeConfig. This method needed to be called before creating the cache and before route matching. 

I moved the latter to the KernelEvent Booted to seperate the route matching from the route discovery as much as possible.